wrappers/cabal.hs

```diff
@@ -10,9 +10,11 @@ main = do
   case args of
     "--interactive":_ -> do
       getCurrentDirectory >>= putStrLn
-      -- note this probably breaks if paths have spaces in
-      putStrLn $ unwords args
+      putStrLn $ delimited args
     _ -> do
       ph <- spawnProcess "ghc" args
       code <- waitForProcess ph
       exitWith code
+
+delimited :: [String] -> String
+delimited = (++) "-here " . foldr (\a as -> a ++ '\NUL' : as) ""
```

Comment:
The `foldr` bit is just `intercalate` in disguise.

Reply:
It's different from `intercalate` because it puts `'\NUL'` characters *after* every list element, instead of *between* elements. `delimited` will put a `'\NUL'` at the end of the list, `intercalate` will not.

Reply:
`concatMap (++ "\NUL")` ? I find the foldr very hard to interpret, hence my desire to drop it!